go-controller/pkg/node/secondary_node_network_controller.go

```diff
@@ -0,0 +1,82 @@
+package node
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	ovntypes "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/types"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/klog/v2"
+)
+
+// SecondaryNodeNetworkController structure is the object which holds the controls for starting
+// and reacting upon the watched resources (e.g. pods, endpoints) for secondary network
+type SecondaryNodeNetworkController struct {
+	BaseNodeNetworkController
+}
+
+// NewSecondaryNodeNetworkController creates a new OVN controller for creating logical network
+// infrastructure and policy for default l3 network
+func NewSecondaryNodeNetworkController(cnnci *CommonNodeNetworkControllerInfo, netInfo util.NetInfo,
+	netconfInfo util.NetConfInfo) *SecondaryNodeNetworkController {
+	return &SecondaryNodeNetworkController{
+		BaseNodeNetworkController: BaseNodeNetworkController{
+			CommonNodeNetworkControllerInfo: *cnnci,
+			NetConfInfo:                     netconfInfo,
+			NetInfo:                         netInfo,
+			stopChan:                        make(chan struct{}),
+			wg:                              &sync.WaitGroup{},
+		},
+	}
+}
+
+// Start starts the default controller; handles all events and creates all needed logical entities
+func (nc *SecondaryNodeNetworkController) Start(ctx context.Context) error {
+	klog.Infof("Start secondary node network controller of network %s", nc.GetNetworkName())
+
+	isOvnUpEnabled := atomic.LoadInt32(&nc.atomicOvnUpEnabled) > 0
+
+	if err := nc.nodeUpgrade(ctx, isOvnUpEnabled, nil, nil, ""); err != nil {
+		return err
+	}
+
+	go wait.Until(func() {
+		nc.checkForStaleOVSRepresentorInterfaces()
+	}, time.Minute, nc.stopChan)
+
+	return nil
+}
+
+// Stop gracefully stops the controller
+func (nc *SecondaryNodeNetworkController) Stop() {
+	klog.Infof("Stop secondary node network controller of network %s", nc.GetNetworkName())
+	close(nc.stopChan)
+	nc.wg.Wait()
+}
+
+// Cleanup cleans up node entities for the given secondary network
+func (nc *SecondaryNodeNetworkController) Cleanup(netName string) error {
```

Comment:
This function call a "removeVFRepresentorPorts", which should call findAllVFRepresentorPorts(name networkName). FindAllVFRepresentorPorts should be leveraged by the checkForStaleOVSRepresentorInterfaces and whatever other fucntion needs to look up vf reps so we are not duplicating code around.

Reply:
this one is very different from checkForStaleOVSRepresentorInterfaces(), the latter is not to delete all VF representor ports of the given network (only those whose Pods are deledted).

Reply:
yeah but your search code can be common, and cleanup doesnt necessarily just only mean VF ports, so that should be a sub-function call. For example:

```
func (nc *SecondaryNodeNetworkController) Cleanup(netName string) error {
   return util.RemoveVFRepsByNetwork(netName)
}

// util package
func RemoveVFRepsByNetwork(netName string) error {
     netIDs := map[string]string{ovntypes.NetworkExternalID: netName}
     portNames, err := FindVFRepsWithExternalIDs(netIDs)
     if err != nil { return err }
     return RemoveOVSPortByName(portNames)
}

func RemoveStaleVFReps() {
     // to be implemented calling other fucntions
}

func FindVFRepsWithExternalIDs(extraIDs map[string]string) ([]string, error) {
    command := []string{"--columns=name", "--data=bare", "--no-headings",
		"--format=csv", "find", "Interface", "external_ids:sandbox!=\"\"", "external_ids:vf-netdev-name!=\"\""}
    for k,v : = range extraIDS{
         command = append(command, fmt.Sprintf("external_ids:%s==%s", k, v))
     }
     out, stderr, err := util.RunOVSVsctl(command...)
	if err != nil {
		return nil, fmt.Errorf("failed to list ovn-k8s OVS interfaces:, stderr: %q, error: %v", stderr, err)
	}
   
     names := strings.Split(out, "\n")
     return names, nil 
}

func RemoveOVSPortByName( names string) error{
        for _, name := range names {
		_, stderr, err := util.RunOVSVsctl("--if-exists", "--with-iface", "del-port", name)
		if err != nil {
			return fmt.Errorf("failed to delete interface %q . stderr: %q, error: %v",
				name, stderr, err)
		}
      }
     return nil
}
```